pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1022,21 +1021,33 @@ pub struct DatadirModification<'a> {
     // The put-functions add the modifications here, and they are flushed to the
     // underlying key-value store by the 'finish' function.
     pending_lsns: Vec<Lsn>,
-    pending_updates: HashMap<Key, Vec<(Lsn, Value)>>,
+    pending_updates: HashMap<Key, Vec<(Lsn, usize, Value)>>,
     pending_deletions: Vec<(Range<Key>, Lsn)>,
     pending_nblocks: i64,
 
     /// For special "directory" keys that store key-value maps, track the size of the map
     /// if it was updated in this modification.
     pending_directory_entries: Vec<(DirectoryKind, usize)>,
+
+    /// An **approximation** of how large our EphemeralFile write will be when committed.
+    pending_bytes: usize,
```

Comment:
It would be approximately track the number of bytes written to ephemeral layers, I'm not sure if we track that anywhere else already.

When we do the data/metadata split, I think it'll be interesting to have a broken-down metric by types of data we're ingesting: ordinary page deltas, images, SLRU deltas, metadata pages -- that will make any pathological workloads stick out.